domain/schema/model.go

```diff
@@ -541,14 +543,40 @@ CREATE TABLE storage_instance_constraint (
 CREATE UNIQUE INDEX idx_storage_instance_constraint
 ON storage_instance_constraint (storage_instance_uuid, constraint_type_id);
 
+CREATE TABLE storage_provisioning_status (
+    id          INT PRIMARY KEY,
+    name        TEXT NOT NULL,
+    description TEXT
+);
+
+CREATE UNIQUE INDEX idx_storage_provisioning_status
+ON storage_provisioning_status (name);
+
+INSERT INTO storage_provisioning_status VALUES
+    (0, 'pending', 'Creation or attachment is awaiting completion'), 
+    (1, 'provisioned', 'Requested creation or attachment has been completed'),
+    (2, 'error', 'An error was encountered during creation or attachment'); 
+
 CREATE TABLE storage_volume (
-    uuid    TEXT PRIMARY KEY,
-    life_id INT NOT NULL,
-    name    TEXT NOT NULL,
-    -- TODO (manadart 2023-12-11) info and params.
+    uuid                   TEXT PRIMARY KEY,
+    life_id                INT NOT NULL,
+    name                   TEXT NOT NULL,
+    provider_id            TEXT,
+    storage_pool_uuid      TEXT,
+    size_mib               INT,
+    hardware_id            TEXT,
+    wwn                    TEXT,
```

Comment:
Is this just for MAAS?

Reply:
It's a shame we can't have a composed table for just maas.

Reply:
We could start instances on other providers the same way as maas - add the block device requirements to the start instance request and get back the block device info in the response. eg we could do this with EC2 with the official AWS SDK if we wanted but when EC2 was first done in juju, we had the home grown SDK which doesn't support it. Pretty sure other providers would be similar.